File: app/src/main/java/io/awaken/ui/connections/ConnectionsAdapter.java
package io.awaken.ui.connections;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import java.util.List;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;
import io.awaken.R;
import io.awaken.data.model.Connection;
import io.awaken.ui.utils.AnimatedRecyclerView;

/**
 * @author Tyler Wong
 */
public class ConnectionsAdapter extends RecyclerView.Adapter<ConnectionViewHolder> {

    private List<Connection> mConnections;
    private AnimatedRecyclerView mRecyclerView;
    private ConnectionRefresher mRefresher;

    ConnectionsAdapter(AnimatedRecyclerView recyclerView, List<Connection> connections, ConnectionRefresher refresher) {
        this.mRecyclerView = recyclerView;
        this.mConnections = connections;
        this.mRefresher = refresher;
    }

    public List<Connection> getConnections() {
        return mConnections;
    }

    public void setConnections(List<Connection> connections) {
        mConnections = connections;
        notifyDataSetChanged();
    }

    @NonNull
    @Override
    public ConnectionViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext())
                .inflate(R.layout.connection_card, parent, false);

        ConnectionViewHolder viewHolder = new ConnectionViewHolder(view, mRecyclerView, mRefresher);
        return viewHolder;
    }

    @Override
    public void onBindViewHolder(@NonNull final ConnectionViewHolder holder, int position) {
        Connection curConnection = mConnections.get(position);

        holder.mNickname.setText(curConnection.getNickname());
        holder.mHost.setText(curConnection.getHost());
        holder.mMac.setText(curConnection.getMac());
        holder.mLocation.setText(curConnection.getCity() + ", " + curConnection.getState());
        holder.mDate.setText(curConnection.getDate());
        boolean status = Boolean.parseBoolean(curConnection.getStatus());
        if (status) {
            holder.mStatus.setImageResource(R.drawable.active_marker);
        } else {
            holder.mStatus.setImageResource(R.drawable.inactive_marker);
        }
        holder.setId(curConnection.getId());
    }

    @Override
    public int getItemCount() {
        return mConnections.size();
    }

    @Override
    public void onAttachedToRecyclerView(@NonNull RecyclerView recyclerView) {
        super.onAttachedToRecyclerView(recyclerView);
    }
}

File: app/src/main/java/io/awaken/ui/connections/ConnectionsAdapter.kt
package io.awaken.ui.connections

import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import io.awaken.R
import io.awaken.data.model.Connection

typealias ConnectionsRefreshListener = () -> Unit

internal class ConnectionsAdapter(
        private val refreshListener: ConnectionsRefreshListener
) : RecyclerView.Adapter<ConnectionViewHolder>() {

    private var connections = listOf<Connection>()

    fun setConnections(connections: List<Connection>) {
        this.connections = connections
        notifyDataSetChanged()
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ConnectionViewHolder {
        val view = LayoutInflater.from(parent.context)
                .inflate(R.layout.connection_card, parent, false)

        return ConnectionViewHolder(view, refreshListener)
    }

    override fun onBindViewHolder(holder: ConnectionViewHolder, position: Int) {
        val curConnection = connections[position]

        holder.nickname.text = curConnection.nickname
        holder.host.text = curConnection.host
        holder.mac.text = curConnection.mac
        holder.location.text = curConnection.city + ", " + curConnection.state
        holder.date.text = curConnection.date
        val status = curConnection.status.toBoolean()
        if (status) {
            holder.status.setImageResource(R.drawable.active_marker)
        } else {
            holder.status.setImageResource(R.drawable.inactive_marker)
        }
        holder.connectionId = curConnection.id
    }

    override fun getItemCount() = connections.size
}

